lib/logger_humio_backend.ex

```diff
@@ -11,40 +11,59 @@ defmodule Logger.Backend.Humio do
   @default_client Logger.Backend.Humio.Client.Tesla
   @default_level :debug
   @default_metadata []
-  @default_max_batch_size 50
+  @default_max_batch_size 20
+  @default_flush_interval_ms 10_000
 
   @type log_event :: %{
           level: atom(),
           message: String.t(),
           timestamp: any(),
           metadata: keyword()
         }
-  @type log_events :: [log_event]
   @type state :: %{
-          log_events: log_events,
-          config: map()
+          log_events: [log_event],
+          config: %{
+            token: String.t(),
```

Comment:
Would it maybe make sense to use `required()` for the map typespec? ie:
```suggestion
            required(:token) => String.t(),
```

Reply:
I think that if there's really a need to enforce a required field, then the state should actually be a struct rather than a map

Reply:
If it's actually required, maybe it's better to use a struct for the state